src/routes/_surfaces/workspace/explore/time-series-charts/MetricsTimeSeriesCharts.svelte

```diff
@@ -43,16 +39,32 @@
 
   let mouseoverValue = undefined;
 
-  function initializeToMidnight(dt) {
-    let newDt = new Date(dt);
-    newDt.setHours(0, 0, 0, 0);
-    return newDt;
+  $: key = `${startValue}` + `${endValue}`;
+
+  function getMinTs(values: TimeSeriesValue[]): Date {
+    if (!values) return new Date();
+    let min = new Date(values[0].ts);
+    for (let i = 1; i < values.length; i++) {
+      if (new Date(values[i].ts).getTime() < min.getTime()) {
+        min = new Date(values[i].ts);
+      }
+    }
+    return min;
   }
-
-  $: key = `${start}` + `${end}`;
-
-  $: startValue = initializeToMidnight(new Date(start));
-  $: endValue = new Date(end);
+  function getMaxTs(values: TimeSeriesValue[]): Date {
+    if (!values) return new Date();
+    let max = new Date(values[0].ts);
```

Comment:
these could be replaced by d3's `max` and `min` functions from `d3-array`, which we're already using on the data viz side:

```
import { max, min } from 'd3-array'
```

This make it really easy to do these kinds of operations.

https://github.com/d3/d3-array#max

Reply:
Sweet, hadn't used `d3-array` before. Just found the `extent` function which gets both the `min` and `max` in one go.